ethereum_history_api/circuits/lib/src/rlp/types_test.nr

```diff
@@ -49,40 +76,160 @@ mod rlp_fragment {
-        ];
+        global byte = 1;
+        global rlp = [0x01];
+        global fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
 
-        global fragment = RlpFragment { offset: 0, length: 32, data_type: STRING };
+        #[test]
+        fn succes() {
+            fragment.assert_eq_u1("Field", rlp, byte)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            let rlp = [0x01, 0x01];
+            let invalid_length_fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
+
+            invalid_length_fragment.assert_eq_u1("Field", rlp, byte)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+        fn invalid_value() {
+            fragment.assert_eq_u1("Field", rlp, 0)
+        }
+    }
+
+    mod assert_eq_u8 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0x11];
+        global fragment = RlpFragment { offset: 0, length: 1, data_type: STRING };
+        global value = 17;
 
         #[test]
         fn success() {
-            fragment.assert_eq_bytes32("Field", bytes, bytes)
+            fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP length")]
         fn invalid_length() {
+            let rlp = [0x01, 0x01];
             let invalid_length_fragment = RlpFragment { offset: 0, length: 2, data_type: STRING };
 
-            invalid_length_fragment.assert_eq_bytes32("Field", bytes, bytes)
+            invalid_length_fragment.assert_eq_u8("Field", rlp, value)
         }
 
         #[test(should_fail_with = "Field: Invalid RLP value")]
         fn invalid_value() {
-            fragment.assert_eq_bytes32("Field", bytes, alter_array(bytes))
+            fragment.assert_eq_u8("Field", rlp, 16)
+        }
+    }
+
+    mod assert_eq_u32 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0xEE, 0x6B, 0x28, 0x00]; //4000000000
+        global fragment = RlpFragment { offset: 0, length: 4, data_type: STRING };
+        global value = 4000000000;
+
+        #[test]
+        fn success() {
+            fragment.assert_eq_u32("Field", rlp, value)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP length")]
+        fn invalid_length() {
+            fragment.assert_eq_u32("Field", rlp, 0xEE6B28)
+        }
+
+        #[test(should_fail_with = "Field: Invalid RLP value")]
+        fn invalid_value() {
+            fragment.assert_eq_u32("Field", rlp, 0x12345678)
+        }
+    }
+
+    mod assert_eq_u64 {
+        use crate::rlp::{decode::STRING, types::RlpFragment};
+
+        global rlp = [0x8A, 0xC7, 0x23, 0x04, 0x89, 0xE8, 0x00, 0x00];
+        global fragment = RlpFragment { offset: 0, length: 8, data_type: STRING };
+        global value = 10000000000000000000;
```

Comment:
If you use smaller values - tests will be more readable.
Or if you use smth like 0x1000000000